jaworski_cellseg/load_data_widget.py

```diff
@@ -2,19 +2,83 @@
 import napari
 from pathlib import Path
 from bioio import BioImage
+import tifffile
+import numpy as np
 
 
-def jwslab_load_bio_data(napari_viewer: napari.Viewer, file_path: Path):
-    print("enter jwslab_pre_process_file")
+def jwslab_load_bio_data(
+    napari_viewer: napari.Viewer, file_path: Path, physical_sizes: dict
+):
     try:
-        image = BioImage(file_path)
-        bio_data = image.get_image_data("CZYX")
-        napari_viewer.add_image(
-            bio_data,
-            channel_axis=0,
-            name=[f"Channel {i}" for i in range(bio_data.shape[0])],
-        )
-        return bio_data, image.metadata
+        file_extension = file_path.suffix.lower()
+        if file_extension == ".tiff" or file_extension == ".tif":
+            # Load TIFF file using tifffile
+            bio_data = tifffile.imread(file_path)
+            transformed_data = np.moveaxis(bio_data, 1, 0)
+            napari_viewer.add_image(
+                transformed_data,
+                channel_axis=0,
+                name=[f"Channel {i}" for i in range(bio_data.shape[0])],
+            )
+            ## get physical size of the data
+            with tifffile.TiffFile(file_path) as tiff:
+                page = tiff.pages[0]
+                # Extract resolution-related tags
+                x_resolution = page.tags.get("XResolution", None)
+                y_resolution = page.tags.get("YResolution", None)
+
+                unit_name = None
+                z_physical_size = 1.0
+                if tiff.imagej_metadata:
+                    unit_name = tiff.imagej_metadata["unit"]
+                    if "spacing" in tiff.imagej_metadata:
+                        z_physical_size = tiff.imagej_metadata["spacing"]
```

Comment:
Use the get method using those as defaults.
```suggestion
                if tiff.imagej_metadata:
                    unit_name = tiff.imagej_metadata.get("unit")
                    z_physical_size = tiff.imagej_metadata.get("spacing", 1.0)
                else:
                    unit_name = None
                    z_physical_size = 1.0
```

Reply:
Done